src/client/game/answer-input.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+
+import style from '../index.scss';
+
+const SPACE_KEY = 32
+  , ENTER_KEY = 13
+  , UP_KEY = 38
+  , DOWN_KEY = 40;
+
+export default class AnswerInput extends Component {
+    componentWillMount() {
+        this.setState({
+            answerList: [''],
+            focusIndex: 0
+        });
+    }
+
+    handleEnterPress(event) {
+        switch (event.which) {
+            case ENTER_KEY: { //Enter key press
+                if(event.target.value.length > 0) {
+                    let answerList = this.state.answerList;
+                    for(let i = 0; i < answerList.length; i++) {
+                        if(answerList[i] === '') {
+                            this.setState({
+                                answerList: [...this.state.answerList],
+                                focusIndex: i
+                            });
+                            return;
+                        }
+                    }
+                    this.setState({
+                        answerList: [...this.state.answerList, ''],
+                        focusIndex: this.state.answerList.length
+                    });
+                }
+                break;
+            }
+            case UP_KEY: {
+                let focusIndex = this.state.focusIndex === 0 ? 0 : this.state.focusIndex-1;
+                this.setState({
+                    focusIndex: focusIndex
+                });
+                break;
+            }
+            case DOWN_KEY: {
+                let focusIndex = this.state.focusIndex === this.state.answerList.length-1 ? this.state.answerList.length-1 : this.state.focusIndex+1;
+                this.setState({
+                    focusIndex: focusIndex
+                });
+                break;
+            }
+            case SPACE_KEY: {
+                event.preventDefault();
+                break;
+            }
+        }
+    }
+
+    handleChange(event, id) {
+        this.setState({
+            answerList: this.state.answerList
+                .map((originalAnswer, index) => {
+                    if(index === id) {
+                        return event.target.value;
+                    }
+                    return originalAnswer;
+                })
+        });
+    }
+    handleFocus(index) {
+        this.setState({
+            focusIndex: index
+        });
+    }
+
+    textBoxGenerator(props, focusIndex) {
```

Comment:
props? Should be answer list I suspect. Also - both these values are coming from the state and as we're in the component that's accessible in here. Use `const { answerList, focusIndex } = this.state;` and then you can remove these parameters